prov/psm/src/psmx_fabric.c

```diff
@@ -103,10 +100,23 @@ int psmx_fabric(struct fi_fabric_attr *attr,
 	if (!fabric_priv)
 		return -FI_ENOMEM;
 
+	psmx_get_uuid(fabric_priv->uuid);
+	if (psmx_env.name_server) {
+		fabric_priv->util_fabric.name_server =
+			calloc(1, sizeof(*fabric_priv->util_fabric.name_server));
+		if (!fabric_priv->util_fabric.name_server) {
+			free(fabric_priv);
+			return -FI_ENOMEM;
+		}
+		fabric_priv->util_fabric.name_server->ns_port =
+			psmx_uuid_to_port(fabric_priv->uuid);
+	}
```

Comment:
that can be tricky. a different way need to be found to pass ns_port information if ns is allocated inside ns_start().